service/src/test/java/uk/gov/hmcts/reform/fpl/utils/ConfidentialDetailsHelperTest.java

```diff
@@ -0,0 +1,58 @@
+package uk.gov.hmcts.reform.fpl.utils;
+
+import org.junit.jupiter.api.Test;
+import uk.gov.hmcts.reform.fpl.model.Address;
+import uk.gov.hmcts.reform.fpl.model.Other;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+
+import java.util.List;
+import java.util.UUID;
+
+import static java.util.UUID.randomUUID;
+import static org.assertj.core.api.Assertions.assertThat;
+import static uk.gov.hmcts.reform.fpl.utils.ConfidentialDetailsHelper.addConfidentialDetailsForElement;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.element;
+
+class ConfidentialDetailsHelperTest {
+
+    private static final String CONFIDENTIAL = "Yes";
+    private static final UUID ID = randomUUID();
+    private static final String TELEPHONE_NUMBER = "01227 831393";
+
+    @Test
+    void shouldAddConfidentialDetailsForOther() {
+        List<Element<Other>> others = List.of(otherWithConfidentialFields(ID, CONFIDENTIAL));
```

Comment:
This isn't testing the correct thing here. It should test that the matching element passed in is found in the confidential list. 

Something like this 

'    @Test
    void shouldFindItemToAddWhenInConfidential() {
         UUID firstUUID = fromString("d733b442-1b70-4abb-87b6-47e406d2e32a");
         UUID secondUUID = fromString("d733b442-1b70-4abb-87b6-47e406d2e32b");

        List<Element<Other>> confidential = List.of(otherWithConfidentialFields(firstUUID, CONFIDENTIAL, TELEPHONE_NUMBER),
            otherWithConfidentialFields(secondUUID, CONFIDENTIAL, "02830860107"));
        Element<Other> element = otherWithRemovedConfidentialFields(firstUUID);

        Other confidentialOther = addConfidentialDetailsForElement(confidential, element);

        assertThat(confidentialOther).isEqualToComparingFieldByField(confidential.get(0));
    }'

Reply:
Changed test name and modified test to test fieldByField instead of 1 field.